embedded-storage-async/src/nor_flash.rs

```diff
@@ -52,8 +52,12 @@ pub trait NorFlash: ReadNorFlash {
 impl<T: ReadNorFlash> ReadNorFlash for &mut T {
 	const READ_SIZE: usize = T::READ_SIZE;
 
-	async fn read(&mut self, offset: u32, bytes: &mut [u8]) -> Result<(), Self::Error> {
-		T::read(self, offset, bytes)
+	async fn read(
+		&mut self,
+		offset: u32,
+		bytes: &mut [u8],
+	) -> Result<(), <&mut T as ErrorType>::Error> {
```

Comment:
if you enable `#![feature(impl_trait_projections)]` then you can do `Self::Error` instead of this awkward thing.

(I think `impl_trait_projections` is likely to get stabilized together with AFIT, since it's necessary for basic usage, so I wouldn't worry about depending on it)